src/transformers/models/rt_detr/convert_rt_detr_original_pytorch_checkpoint_to_pytorch.py

```diff
@@ -0,0 +1,305 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Convert RT Detr checkpoints with Timm backbone"""
+
+import argparse
+from pathlib import Path
+
+import requests
+import torch
+from PIL import Image
+
+from transformers import RTDetrConfig, RTDetrImageProcessor, RTDetrModel
+
+
+replaces = {}
+
+# Mapping layers
+conv_layers = [("conv1_1", "conv1.0"), ("conv1_2", "conv1.3")]
+layers_to_map = ["conv.weight", "norm.weight", "norm.bias", "norm.running_mean", "norm.running_var"]
+
+for origin, dest in conv_layers:
+    dest_order = int(dest[-1])
+    versions = [dest_order if i == 0 else dest_order + 1 for i in range(len(layers_to_map))]
+    for idx, layer in enumerate(layers_to_map):
+        layer_name, layer_attribute = layer.split(".")
+        key = f"backbone.conv1.{origin}.{layer}"
+        val = f"model.backbone._backbone.{dest[:-1]}{versions[idx]}.{layer_attribute}"
+        replaces[key] = val
+
+# Mapping layers with batch norm
+conv_layers = ["conv{}.weight", "bn{}.weight", "bn{}.bias", "bn{}.running_mean", "bn{}.running_var"]
+for res_layer in [0, 1, 2, 3]:
+    if res_layer in (0, 3):
+        blocks = [0, 1, 2]
+    elif res_layer == 1:
+        blocks = [0, 1, 2, 3]
+    elif res_layer == 2:
+        blocks = [0, 1, 2, 3, 4, 5]
+    for block in blocks:
+        for id_branch, branch in enumerate(["2a", "2b", "2c"]):
+            for conv_layer, layer in zip(conv_layers, layers_to_map):
+                key = f"backbone.res_layers.{res_layer}.blocks.{block}.branch{branch}.{layer}"
+                val_end = conv_layer.format(id_branch + 1)
+                val = f"model.backbone._backbone.layer{res_layer+1}.{block}.{val_end}"
+                replaces[key] = val
+
+# Mapping conv1_3
+for origin, dest in zip(layers_to_map, conv_layers):
+    key = f"backbone.conv1.conv1_3.{origin}"
+    version = "1.6" if dest.startswith("conv") else "1"
+    val = f"model.backbone._backbone.{dest.format(version)}"
+    replaces[key] = val
+
+# Mapping downsample layers
+for layer_number in range(0, 4):
+    extra_conv = ".conv" if layer_number != 0 else ""
+    for id, layer_name in enumerate(layers_to_map):
+        downsample = "1" if id == 0 else "2"
+        end_layer = layer_name[layer_name.find(".") + 1 :]
+        key = f"backbone.res_layers.{layer_number}.blocks.0.short{extra_conv}.{layer_name}"
+        val = f"model.backbone._backbone.layer{layer_number+1}.0.downsample.{downsample}.{end_layer}"
+        replaces[key] = val
+
+# This mapping is different from models trained in COCO (e.g. Detr)
+id2label = {
```

Comment:
to verify: where's the class "no object" being mapped? config.num_labels + 1?

Reply:
For DETR, I manually added a +1 to add the "no object" class: https://github.com/huggingface/transformers/blob/35551f9a0f66a22de4971b4a51b3c172d3b87f95/src/transformers/models/detr/modeling_detr.py#L1480. This was also done like so in the original implementation.

However, alternatively you could add the "no object" class in the `id2label`, which will make sure num_labels includes one additional class.

Reply:
the problem of adding an extra class in the `id2label` (e.g. `91: "no object"`) is that the current checkpoint may not be able to output `91`. I need to investigate how RTDetr deals with "no object"

Reply:
@rafaelpadilla are you still looking to merge this new model? on a side note the no_object got RTDetr in the original repo is not included https://github.com/lyuwenyu/RT-DETR/blob/3330eca679a7d7cce16bbb10509099174a2f40bf/rtdetr_pytorch/src/zoo/rtdetr/matcher.py#L23